### DISCUSSION TITLE
Is it recommened to use `--watch-config` in production?

### DISCUSSION THREAD

**Discussion by pgollangi** [vectordotdev/vector#20705] [Q&A] [answered]
Hi,

I have a use case to dynamically populate metadata to logs and route to different syncs. We can probably do it by dynamically updating configuration that transform and route and use `--watch-config` global option on vector to reload when configuration is changed.

I have following questions of `--watch-config` and `reload`ing of vector and its performance impact:

1. Is it recommend to use `--watch-config` in production where vector handling 1M msg/s
2. how intensive is `reload` of vector, length of configuration file has a factor on this?
3. will there be a downtime or dip in # of events that vector accepts when its reloading?
4. vector `reload` cause any back pressure on systems writing to it?
5. Is it possible that vector discord events during reload?

Also, is there a limit on # of routes / transform in config for optimal vector performance

Thanks for the help!

**Comment by jszwedko** [comment, marked as answer]
> * Is it recommend to use `--watch-config` in production where vector handling 1M msg/s

You could try it. It should work, ideally, but at that volume I wouldn't be surprised if some edge case was hit.

> * how intensive is `reload` of vector, length of configuration file has a factor on this?

No more intensive than loading the file at start-up. And yes, it depends on the number of components.

> * will there be a downtime or dip in # of events that vector accepts when its reloading?

Ideally no.

> * vector `reload` cause any back pressure on systems writing to it?

Ideally no.

> * Is it possible that vector discord events during reload?

The expectation is that it shouldn't. If it does, that would be a bug.

**Reply by pgollangi** [reply]
Thank you for taking time to answer!
